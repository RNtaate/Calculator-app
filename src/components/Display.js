import React from 'react';

let Display = () => {
  return(
    <input type="text"/>
  );
}

export default Display;