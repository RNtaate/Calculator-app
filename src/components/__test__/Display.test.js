import React from 'react';
import ReactDOM from 'react-dom';
import Display from '../Display';

import {render} from '@testing-library/react';

import renderer from 'react-test-renderer';

it( 'renders display compoonent without crashing', () => {
  const myDiv = document.createElement('div');
  ReactDOM.render(<Display />, myDiv);
});

it('renders the display component correctly', () => {
  const{getByRole} = render(<Display />);
  expect(getByRole('textbox')).toHaveDisplayValue('0');
});

it('renders the display component correctly', () => {
  const{getByRole} = render(<Display />);
  expect(getByRole('textbox')).toHaveAttribute('value', '0');
});

it('renders the display component correctly when using props', () => {
  const{getByRole} = render(<Display result="10"/>);
  expect(getByRole('textbox')).toHaveDisplayValue('10');
});

it('matches the Display component snapshot', () => {
  const tree = renderer.create(<Display />);
  expect(tree).toMatchSnapshot();
});

it('matches the Display component snapshot', () => {
  const tree = renderer.create(<Display result="1994"/>);
  expect(tree).toMatchSnapshot();
});