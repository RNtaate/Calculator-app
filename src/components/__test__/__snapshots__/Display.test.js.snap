// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the Display component snapshot 1`] = `
<input
  className="display-input"
  readOnly={true}
  style={
    Object {
      "textAlign": "right",
    }
  }
  type="text"
  value="0"
/>
`;

exports[`matches the Display component snapshot 2`] = `
<input
  className="display-input"
  readOnly={true}
  style={
    Object {
      "textAlign": "right",
    }
  }
  type="text"
  value="1994"
/>
`;
